zero_bin/rpc/src/native/txn.rs

```diff
@@ -357,3 +398,228 @@ fn prestate_tracing_options(diff_mode: bool) -> GethDebugTracingOptions {
+                    .stack
+                    .as_ref()
+                    .expect("No stack found in structLog.")
+                    .iter()
+                    .rev();
+                let jump_target = evm_stack
+                    .next()
+                    .expect("Stack must contain at least one value for a JUMP instruction.")
+                    .to::<u64>();
+                prev_jump = Some(jump_target);
+            }
+            "JUMPI" => {
+                let mut evm_stack = entry
+                    .stack
+                    .as_ref()
+                    .expect("No stack found in structLog.")
+                    .iter()
+                    .rev();
+                let jump_target = evm_stack
+                    .next()
+                    .expect("Stack must contain at least one value for a JUMP instruction.")
+                    .to::<u64>();
+                let jump_condition = evm_stack
+                    .next()
+                    .expect("Stack must contain at least two values for a JUMPI instruction.")
+                    .is_zero()
+                    .not();
+
+                prev_jump = if jump_condition {
+                    Some(jump_target)
+                } else {
+                    None
+                };
+            }
+            "JUMPDEST" => {
+                let (code_hash, ctx) = call_stack
+                    .last()
+                    .expect("Call stack was empty when a JUMPDEST was encountered.");
+                let jumped_here = if let Some(jmp_target) = prev_jump {
+                    debug_assert!(
+                        jmp_target == entry.pc,
+                        "The structlog seems to make improper JUMPs."
+                    );
+                    jmp_target == entry.pc
+                } else {
+                    false
+                };
+                let offset = &(entry.pc as usize);
+                if jumped_here {
+                    jumpdest_table
+                        .0
+                        .entry(*code_hash)
+                        .or_insert(ContextJumpDests::default());
+
+                    jumpdest_table
+                        .0
+                        .get_mut(code_hash)
+                        .unwrap()
+                        .0
+                        .entry(*ctx)
+                        .or_insert(BTreeSet::default());
+
+                    jumpdest_table
+                        .0
+                        .get_mut(code_hash)
+                        .unwrap()
+                        .0
+                        .get_mut(ctx)
+                        .unwrap()
+                        .insert(*offset);
+
+                    debug_assert!(jumpdest_table.0.contains_key(code_hash));
+                    debug_assert!(jumpdest_table.0[code_hash].0.contains_key(ctx));
+                    debug_assert!(jumpdest_table.0[code_hash].0[ctx].contains(offset));
+                }
```

Comment:
I assumed this will go away post debugging?

Reply:
Assertion yes.  I have refactored it into a method.